src/mds/MDBalancer.cc

```diff
@@ -1404,11 +1405,23 @@ void MDBalancer::handle_mds_failure(mds_rank_t who)
   }
 }
 
-int MDBalancer::dump_loads(Formatter *f) const
+int MDBalancer::dump_loads(Formatter *f, const cmdmap_t &cmdmap) const
 {
-  std::deque<CDir*> dfs;
+  int64_t depth = 0;
+  bool got = ceph::common::cmd_getval(cmdmap, "depth", depth);
+  if (!got || depth <= 0) {
+    dout(10) << "no depth limit when dirfrags dump_load" << dendl;
+  }
```

Comment:
please move this parsing and error reporting to `MDSRankDispatcher::handle_asok_command()` and only pass the `int` value of `depth` to this function